### PR TITLE
[Core] Migrate `ManagerInterface` `identifier` and `displayName` to C++

### PR DESCRIPTION
**Pull request by feltech**
For #292

### CODE REVIEW

**Comment by feltech** [comment]
Force-pushes up to here would have been avoided if we had #251 done :slightly_frowning_face:

File: doc/src/Testing.dox [hunk 1]
```diff
@@ -28,7 +28,7 @@
  * values to pass to methods such as
  * @ref openassetio.managerAPI.ManagerInterface.ManagerInterface.isEntityReference
  * "isEntityReference", and the expected values from methods such as
- * @ref openassetio.managerAPI.ManagerInterface.ManagerInterface.identifier
+ * @ref openassetio::v0_0::managerAPI::ManagerInterface::identifier
```
Comment (foundrytom) [inline doc/src/Testing.dox:31]: [consider] Do we need to create some Doxygen constant/macro/something so we dont have to keep `s/vX_X/vY_Y/g` every time we change the version?

**Review by foundrytom — COMMENTED** [review]
Sorry only had time for a quick skim, but looks pretty good to me.

**Review by fnRaihanKibria — APPROVED** [review]
LGTM

File: doc/src/Testing.dox (hunk 1 above)
Comment (feltech) [reply doc/src/Testing.dox:31]: The only solution I've found for string substitution is `$(...)` environment variable substitution.  

Annoyingly, although env vars are substituted in `@ref`s, the substitution is not performed before attempting to cross-reference, so the `@ref` fails, but the rendered output has the correct namespace. I.e.
```
 @ref openassetio::$(OPENASSETIO_VERSION)::managerAPI::ManagerInterface::identifier
```
emits an "unable to resolve reference to `openassetio:'" warning and renders as
```
openassetio::v0_0::managerAPI::ManagerInterface::identifier "identifier"
```

An option that works, though may be controversial, is to add another `ALIAS`, which I'll call `fqref` (fully-qualified ref) and looks like
```
@fqref{managerAPI::ManagerInterface::identifier} "identifier"
```
It seems the `{...}` is then required, unfortunately.  However, I kinda like that it removes the prefix cruft.

Comment (foundrytom) [reply doc/src/Testing.dox:31]: Good hunting. `@fref` looks good to me.

Comment (feltech) [reply doc/src/Testing.dox]: Added/used `@fqref` in 171ca60c

Comment (foundrytom) [reply doc/src/Testing.dox]: https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/295/commits/171ca60cc1a69aaffefe035e634c9b0714163939 = 🔥

**Review by foundrytom — APPROVED** [review]

**Comment by foundrytom** [comment]
@feltech feel free to squash and merge at your leisure.

File: doc/src/Testing.dox (hunk 1 above)
Comment (feltech) [reply doc/src/Testing.dox]: I feel sorry of @ref (though their repo is empty, probably for good reason)

**Pull request merged**